src/main/java/org/mifos/connector/mockPaymentSchema/service/BatchServiceImpl.java

```diff
@@ -0,0 +1,163 @@
+package org.mifos.connector.mockPaymentSchema.service;
+
+import org.mifos.connector.mockPaymentSchema.schema.BatchDTO;
+import org.mifos.connector.mockPaymentSchema.schema.BatchDetailResponse;
+import org.mifos.connector.mockPaymentSchema.schema.Transfer;
+import org.mifos.connector.mockPaymentSchema.schema.TransferStatus;
+import org.springframework.stereotype.Service;
+
+import java.math.BigDecimal;
+import java.util.*;
+
+@Service
+public class BatchServiceImpl implements BatchService {
+
+    private List<String> requestIds;
+
+    private List<String> workflowInstanceKeys;
+
+    private String payerPartyId = "835322416";
+
+    private String payeePartyId = "27713803912";
+
+    private int successTxnCount = 9;
+
+
+
+    public BatchServiceImpl(){
+        requestIds = new ArrayList<>(Arrays.asList("f1e22fe3-9740-4fba-97b6-78f43bfa7f2f",
+                "39f6ac4d052e-72aa3ea4-e6f6-4880-877f", "a27631f6-6dd4-4d69-b4fc-8932bd721913",
+                "3d21e6ea-c583-44ed-b94f-af909fa7616e", "15f9a0b0-2299-436d-8433-da564140ba66",
+                "f1e22fe3-9740-4fba-97b6-78f49bfa7f2f", "39f6ac4d052e-72aa3ea4-e6f6-4380-877f",
+                "a27631f6-6dd4-4d69-b4fc-8452bd721913", "3d21e6ea-c583-44ed-b94f-af909fu7616e",
+                "15f9a0b0-2299-436d-8433-da667140ba66"));
+
+        workflowInstanceKeys = new ArrayList<>(Arrays.asList("1257787875372524", "3265012419947587", "9816176217118279",
+                "7686876624590145", "7089769829361117", "6120480408758686", "6313003282228195", "6310718206541276",
+                "3936102393207027", "6460724468126500"));
+    }
+
+    @Override
+    public BatchDTO getBatchSummary(String batchId) {
+        Random random = new Random();
+        int randomValue = random.nextInt(2) + 1;
+
+        if(randomValue==1){
+            return successfulBatchSummaryResponse(batchId);
+        }
+
+        return failureBatchSummaryResponse(batchId);
+    }
+
+    @Override
+    public BatchDetailResponse getBatchDetails(String batchId, int pageNo, int pageSize) {
+
+
+        List<Transfer> transactions = getTransactions(batchId);
+        int toIndex = pageNo * pageSize;
+        int fromIndex = toIndex - pageSize;
+        toIndex = Math.min(toIndex, transactions.size());
+        BatchDetailResponse batchDetailResponse = new BatchDetailResponse();
+        batchDetailResponse.setContent(transactions.subList(fromIndex, toIndex));
+        return batchDetailResponse;
+    }
+
+    private BatchDTO failureBatchSummaryResponse(String batchId) {
+        BatchDTO batchDTO = new BatchDTO();
+
+        batchDTO.setBatch_id(batchId);
```

Comment:
batch DTO already has couple of constructors added, why do we need to set the params individually?